### ISSUE TITLE
E3002: Invalid Property Resources/Stackset/Properties/OperationPreferences/ConcurrencyMode

### ISSUE THREAD

**Issue by devvick** [aws-cloudformation/cfn-lint#3459]
### CloudFormation Lint Version

0.87.9

### What operating system are you using?

Archlinux - package from official repos

### Describe the bug

AWS::CloudFormation::StackSet OperationPreferences should allow ConcurrencyMode
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cloudformation-stackset-operationpreferences.html

### Expected behavior

`ConcurrencyMode` is a property launched in November 2023. cfn-lint should validate and allow this property.

### Reproduction template

```yaml
AWSTemplateFormatVersion: "2010-09-09"

Resources:
  Stackset:
    Type: AWS::CloudFormation::StackSet
    Properties:
      StackSetName: myStackset
      PermissionModel: SERVICE_MANAGED
      CallAs: DELEGATED_ADMIN
      Capabilities:
        - CAPABILITY_IAM
      AutoDeployment:
        Enabled: true
        RetainStacksOnAccountRemoval: false
      ManagedExecution:
        Active: true
      OperationPreferences:
        ConcurrencyMode: SOFT_FAILURE_TOLERANCE
        FailureToleranceCount: 2
        MaxConcurrentCount: 3
        RegionConcurrencyType: PARALLEL
      TemplateURL: stackset.yml
```

**Comment by kddejong** [comment]
We will have to patch it in as it isn't provided in the specs we get from [here](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-resource-specification.html)

**Comment by kddejong** [comment]
Problem is fixed in the v0 branch and will be included in the next release.
